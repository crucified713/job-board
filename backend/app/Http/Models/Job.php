<?php

namespace App\Http\Models;

use Illuminate\Database\Eloquent\Model;

class Job extends Model
{
  /**
   * The attributes that are mass assignable.
   *
   * @var array
   */
  protected $fillable = [
    'title', 'description', 'location',
  ];

  /**
   * The attributes excluded from the model's JSON form.
   *
   * @var array
   */
  protected $hidden = [];
}