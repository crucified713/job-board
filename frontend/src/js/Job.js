import React, { Component } from 'react';

class Job extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isCollapsed: true
    }
    this.toggleDetail = this.toggleDetail.bind(this);
  }

  toggleDetail() {
    this.setState((prevState) => {
      return {isCollapsed: !prevState.isCollapsed}
    });
  }

  render() {
    return (
      <article>
        <h5 className="jobs__title" onClick={this.toggleDetail}>{(this.state.isCollapsed ? '+ ' : '- ') + this.props.title}</h5>
        <div className={"jobs__collapse " + (this.state.isCollapsed ? "" : "jobs__collapse--show")}>

          <p><small><strong>Location: </strong>{this.props.location ? this.props.location : 'No locations'}</small></p>
          <br/>
          <p>{this.props.description ? this.props.description : 'No description'}</p>
          <br />
        </div>
      </article>
    );
  }
}

export default Job;