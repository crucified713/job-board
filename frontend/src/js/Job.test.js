import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import Job from './Job';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Job', function () {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Job title="Admin" location="Melbourne" description="Some description"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should update state on click', function() {
    const wrapper = shallow(<Job title="Admin" location="Melbourne" description="Some description"/>),
      heading = wrapper.find('.jobs__title').first(),
      prevIsCollapsed = wrapper.instance().state.isCollapsed;

    heading.simulate("click");
    wrapper.update();
    expect(wrapper.instance().state.isCollapsed).toBe(!prevIsCollapsed);
  })

});