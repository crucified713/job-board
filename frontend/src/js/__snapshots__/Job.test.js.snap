// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job renders correctly 1`] = `
<article>
  <h5
    className="jobs__title"
    onClick={[Function]}
  >
    + Admin
  </h5>
  <div
    className="jobs__collapse "
  >
    <p>
      <small>
        <strong>
          Location: 
        </strong>
        Melbourne
      </small>
    </p>
    <br />
    <p>
      Some description
    </p>
    <br />
  </div>
</article>
`;

exports[`renders correctly 1`] = `
<article>
  <h5
    className="jobs__title"
    onClick={[Function]}
  >
    + Admin
  </h5>
  <div
    className="jobs__collapse "
  >
    <p>
      <small>
        <strong>
          Location: 
        </strong>
        Melbourne
      </small>
    </p>
    <br />
    <p>
      Some description
    </p>
    <br />
  </div>
</article>
`;
